x/incentives/keeper/distribute.go

```diff
@@ -292,6 +292,12 @@ func (k Keeper) distributeInternal(
 		remainEpochs = gauge.NumEpochsPaidOver - gauge.FilledEpochs
 	}
 
+	// defense in depth
+	// this should never happen in practice since gauge passed in should always be an active gauge.
+	if remainEpochs == uint64(0) {
+		return nil, fmt.Errorf("gauge with id of %d is not active", gauge.Id)
+	}
+
```

Comment:
this should never happen in practice, but if it does, this can lead to chain halt since this would lead to division by zero(=panic) in Begin block, which can potentially lead to chain halt. Thus adding defense in depth